charts/consul/templates/_helpers.tpl

```diff
@@ -15,8 +15,23 @@ as well as the global.name setting.
 {{- end -}}
 {{- end -}}
 
+
 {{- define "consul.restrictedSecurityContext" -}}
 {{- if not .Values.global.enablePodSecurityPolicies -}}
+{{/*
+To be compatible with the 'restricted' Pod Security Standards profile, we
+should set this securityContext on containers whenever possible.
+
+In OpenShift < 4.11 the restricted SCC disallows setting most of these fields,
+so we do not set any for simplicity (and because that's how it was configured
+prior to adding restricted PSA support here). In OpenShift >= 4.11, the new
+restricted-v2 SCC allows setting these in the securityContext, and by setting
+them we avoid PSA warnings that are enabled by default.
+
+We use the K8s version as a proxy for the OpenShift version because there is a
+1:1 mapping of versions. OpenShift 4.11 corresponds to K8s 1.24.x.
```

Comment:
I'm not sure of a better way to check for OpenShift version. As far as we know and based on [their docs](https://access.redhat.com/solutions/4870701) (requires login 😞 ), there's a 1:1 mapping between OpenShift and Kube versions so I think this should be okay.


OpenShift Version | Kubernetes Version
-- | --
4.9 | 1.22
4.10 | 1.23
4.11 | 1.24
4.12 | 1.25
4.13 | 1.26

Reply:
I am also unsure how to unit test this (without actually installing different kubectl / kube versions)

Reply:
Ah, there is `helm template --kube-version`.

```
$ helm template -h | grep kube-version
      --kube-version string                        Kubernetes version used for Capabilities.KubeVersion
```

I'll add a unit test which uses --kube-version to validate this.

Reply:
@pglass added OpenShift versions here for future reference: https://github.com/hashicorp/consul/pull/18307